### PR TITLE
[16.0][MIG] report_xlsx: Migration to 16.0

### PR DESCRIPTION
**Pull request by RodrigoBM**
pr alternative to the original pr https://github.com/OCA/reporting-engine/pull/654

Add this comment https://github.com/OCA/reporting-engine/pull/654#pullrequestreview-1189271013

cc:  @baimont @maq-adhoc @JoanSForgeFlow

### CODE REVIEW

**Review by AntoniRomera — APPROVED** [review]

File: report_xlsx/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            xlsxhttpheaders = [
+                (
+                    "Content-Type",
+                    "application/vnd.openxmlformats-"
+                    "officedocument.spreadsheetml.sheet",
+                ),
+                ("Content-Length", len(xlsx)),
+            ]
+            return request.make_response(xlsx, headers=xlsxhttpheaders)
+        return super(ReportController, self).report_routes(
```
Comment (FernandoRomera) [inline report_xlsx/controllers/main.py]: Apply PEP 3135 (28-Apr-2007) to "super()."

File: report_xlsx/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                            report.print_report_name, {"object": obj, "time": time}
+                        )
+                        filename = "%s.%s" % (report_name, "xlsx")
+                if not response.headers.get("Content-Disposition"):
+                    response.headers.add(
+                        "Content-Disposition", content_disposition(filename)
+                    )
+                return response
+            else:
+                return super(ReportController, self).report_download(data, context)
```
Comment (FernandoRomera) [inline report_xlsx/controllers/main.py]: Apply PEP 3135 (28-Apr-2007) to "super()."

File: report_xlsx/models/ir_report.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        report_model = self.env[report_model_name]
+        return (
+            report_model.with_context(active_model=report_sudo.model)
+            .sudo(False)
+            .create_xlsx_report(docids, data)  # noqa
+        )
+
+    @api.model
+    def _get_report_from_name(self, report_name):
+        res = super(ReportAction, self)._get_report_from_name(report_name)
```
Comment (FernandoRomera) [inline report_xlsx/models/ir_report.py]: Apply PEP 3135 (28-Apr-2007) to "super()."

**Review by FernandoRomera — APPROVED** [review]

File: report_xlsx/controllers/main.py (hunk 1 above)
Comment (RodrigoBM) [reply report_xlsx/controllers/main.py]: Done

File: report_xlsx/controllers/main.py (hunk 2 above)
Comment (RodrigoBM) [reply report_xlsx/controllers/main.py]: Done

File: report_xlsx/models/ir_report.py (hunk 3 above)
Comment (RodrigoBM) [reply report_xlsx/models/ir_report.py]: Done

**Comment by FernandoRomera** [comment]
@pedrobaeza Hi Pedro, can you merge this PR please?

**Comment by alexis-via** [comment]
I think we have a mistake here: in v16, the proto of report_download has changed it is now report_download(self, data, context=None, token=None)

https://github.com/odoo/odoo/blob/16.0/addons/web/controllers/report.py#L90

In report_xlsx, the proto is still report_download(self, data, context=None)
https://github.com/OCA/reporting-engine/blob/16.0/report_xlsx/controllers/main.py#L52

**Comment by alexis-via** [comment]
Here is the kind of warning that you get due to this:
```
2022-12-12 19:51:28,599 62671 WARNING o6_test1 odoo.http: <function odoo.addons.report_xlsx.controllers.main.report_download> called ignoring args {'token'}
```

**Comment by pedrobaeza** [comment]
Can you please fix it or @RodrigoBM?

**Comment by alexis-via** [comment]
ok, I'll try to find time to make a PR

**Comment by alexis-via** [comment]
PR here: https://github.com/OCA/reporting-engine/pull/699

**Pull request merged**